app/src/service/ocr/index.ts

```diff
@@ -1,35 +1,138 @@
+import { createLogger } from "@/service/factory";
+import { type ParserKeys, parseOcrResult, parsers } from "@/service/ocr/parser";
+import {
+  DocumentImage,
+  DocumentOrientation,
+  orientations,
+  rotateDocumentImage,
+} from "@/utils/document";
 import { createWorker } from "tesseract.js";
 
-import { createLogger } from "../../utils/logger";
-import * as parsers from "./parser";
+export type DocumentMatcher<
+  K extends ParserKeys,
+  P extends Record<string, RegExp> = Record<string, RegExp>
+> = {
+  name: string;
+  id: K;
+  patterns: P;
+};
 
-type OcrOptions = {
+export type OcrOptions = {
   debug: boolean;
 };
 
+export type ProcessedImageResult = {
+  documents: ReturnType<typeof parseOcrResult>;
+  image: DocumentImage;
+  percentages: Record<ParserKeys, number>;
+  confidence: number;
+  rotatedOrientation?: DocumentOrientation;
+};
+
+export type ProcessedRotatedImagesResult = ProcessedImageResult[];
+
+export type TextFromImagePathResult = {
+  text: string;
+  confidence: number;
+};
+
 // main logger
 export const logger = createLogger("ocr-parser", {
-  enabled: process.env.LOG_LEVEL === "debug",
+  enabled: false,
   mixin: () => {
     return { module: "ocr-parser" };
   },
 });
 
-const scanImage = async (path: string, opts: OcrOptions) => {
+const getTextFromImagePath = async (
+  document: DocumentImage,
+  opts: OcrOptions
+) => {
   const worker = await createWorker("eng", 1, {
     logger: opts.debug ? (message) => logger.debug(message) : undefined,
   });
-  const result = await worker.recognize(path);
-  const output = result.data.text;
+
+  const result = await worker.recognize(document.data, {
+    rotateAuto: true,
+  });
+
+  const output = {
+    text: result.data.text,
+    confidence: result.data.confidence,
+  };
+
   await worker.terminate();
 
   return output;
 };
 
+// a function that rotates an image and processes it for each orientation
+const processDocument = async (
```

Comment:
I ran out of gas when it came to naming things haha. Do you have any suggestions? Maybe `processImage` which gets called by `processDocument`? Since a single document has 4 different images that should be processed (one for each orientation).